controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -144,6 +144,8 @@
     private static final Set<Action.ActionEffect> WRITE_RUNTIME = EnumSet.of(Action.ActionEffect.WRITE_RUNTIME);
     private static final Set<Action.ActionEffect> ALL_READ_WRITE = EnumSet.of(Action.ActionEffect.READ_CONFIG, Action.ActionEffect.READ_RUNTIME, Action.ActionEffect.WRITE_CONFIG, Action.ActionEffect.WRITE_RUNTIME);
 
+    private static final String UNSUPPORTED_EXPRESSION_PATTERN = ".+\\.\\$\\{.+\\}";
```

Comment:
For any regular expression IMO it is better to add a comment describing how it is structured so if anyone needs to work on it or debug it in the future they can understand the original intent of the regular expression.